src/gui/qgsattributeform.cpp

```diff
@@ -557,7 +555,11 @@ void QgsAttributeForm::updateValuesDependenciesDefaultValues( const int originId
     QgsEditorWidgetWrapper *eww = qobject_cast<QgsEditorWidgetWrapper *>( ww );
     if ( eww )
     {
-      //do not update when when mMode is not AddFeatureMode and it's not applyOnUpdate
+      // Update only on form opening (except when applyOnUpdate is activated)
+      if ( mValuesInitialized && !eww->field().defaultValueDefinition().applyOnUpdate() )
+        continue;
+
+      // Update only when mMode is AddFeatureMode (except when applyOnUpdate is activated)
```

Comment:
Why those are two if conditions and not just one? For the readability?

Reply:
Yes just for readability